core/src/main/java/org/elasticsearch/index/query/functionscore/RandomScoreFunctionBuilder.java

```diff
@@ -122,10 +122,14 @@ protected ScoreFunction doToFunction(QueryShardContext context) {
             // mapper could be null if we are on a shard with no docs yet, so this won't actually be used
             return new RandomScoreFunction();
         }
+        IndexShard indexShard = SearchContext.current().indexShard();
+        /*
+         * This null check works around weirdness in the testing. We don't set indexShard while testing but we need it here. Well, it'd be
+         * better if we could get the shard id out of the context but it isn't available there either.
+         */
         //TODO find a way to not get the shard_id from the current search context? make it available in QueryShardContext?
-        //this currently causes NPE in FunctionScoreQueryBuilderTests#testToQuery
-        final ShardId shardId = SearchContext.current().indexShard().shardId();
-        final int salt = (context.index().getName().hashCode() << 10) | shardId.id();
```

Comment:
I think it still causes NPE, IIRC it's because a search context is not set... so `SearchContext.current().indexShard()` still blows up. The TODO is still valid as a whole... and is the reason why we don't test this function as part of our function score tests